scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java

```diff
@@ -3149,4 +3153,130 @@ public Map<Long, List<String>> getPreciousTaskNames(List<String> jobsIds) {
         }
         return answer;
     }
+
+    public List<JobLabelInfo> getLabels() {
+        return executeReadOnlyTransaction(session -> {
+            logger.info("Loading JobLabels from database");
+
+            Query query = session.getNamedQuery("getTotalJobsLabels").setReadOnly(true);
+            List<Object[]> jobLabelInfoList = query.list();
+            logger.info(jobLabelInfoList.size() + " JobLabels to fetch from database");
+            return jobLabelInfoList.stream()
+                                   .map(label -> new JobLabelInfo(Long.parseLong(label[0].toString()),
+                                                                  label[1].toString()))
+                                   .collect(Collectors.toList());
+        });
+    }
+
+    public List<JobLabelInfo> newLabels(List<String> labels) throws LabelConflictException {
+        List<JobLabelInfo> jobLabelsInfo = new LinkedList<>();
+        for (String label : labels) {
+            if (checkIfLabelExists(label)) {
+                throw new LabelConflictException(label);
+            }
```

Comment:
I think this should be done on all the list before sending sql queries.
Here some labels can be created before reaching an already existing label, which throws the exception and the next labels that can be good will not be created.

So either we reject the whole creations or a silent failure type which i don't think is what we want here

Reply:
I agree. I will refactor this